src/event-stream/tokens-contract-handler.ts

```diff
@@ -626,27 +648,76 @@ class TokensContractHandler {
   /**
    * Make readonly contract call
    */
-  private async makeReadOnlyContractCall(
+  async makeReadOnlyContractCall(
+    functionName: string,
+    functionArgs: ClarityValue[]
+  ): Promise<ReadOnlyContractCallResponse> {
+    const url = this.stacksNetwork.getReadOnlyFunctionCallApiUrl(
+      this.contractAddress,
+      this.contractName,
+      functionName
+    );
+    const args = functionArgs.map(arg => cvToHex(arg));
+
+    const body = JSON.stringify({
+      sender: this.address,
+      arguments: args,
+    });
+
+    let response: Response;
+
+    try {
+      response = await fetchPrivate(url, {
+        method: 'POST',
+        body,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    } catch {
+      // In case there is a network error during fetch.
+      return { found: false, retriable: true };
+    }
+
+    if (!response.ok) {
+      if (response.status >= 500) return { found: false, retriable: true };
+      const msg = await response.text().catch(() => '');
+      return {
+        found: false,
+        retriable: false,
+        error: `Error calling read-only function. Response ${response.status}: ${response.statusText}. Attempted to fetch ${url} and failed with the message: "${msg}"`,
+      };
+    }
+    try {
+      const value = await response.json();
+      return { found: true, value: parseReadOnlyResponse(value) };
+    } catch {
+      // In case there was a connection was interrupted
+      return { found: false, retriable: true };
+    }
+  }
+
+  async retryReadOnlyContractCall(
     functionName: string,
     functionArgs: ClarityValue[]
   ): Promise<ClarityValue> {
-    const txOptions: ReadOnlyFunctionOptions = {
-      senderAddress: this.address,
-      contractAddress: this.contractAddress,
-      contractName: this.contractName,
-      functionName: functionName,
-      functionArgs: functionArgs,
-      network: this.stacksNetwork,
-    };
-    return await callReadOnlyFunction(txOptions);
+    let response: ReadOnlyContractCallResponse = { found: false, retriable: true };
+    while (!response.found && response.retriable) {
+      response = await this.makeReadOnlyContractCall(functionName, functionArgs);
+      // In case of strict mode enabled, retry should be aggresive. Otherwise, queued for later.
+      if (!isMetadataStrictModeEnabled()) await timeout(METADATA_RETREIVAL_INTERVAL);
```

Comment:
Style nit: expression blocks should always have brackets. I.e.
```
if (<condition>) {
  <expression>
}
```
The syntax `if (<condition>) <expression>` is error prone and _should_ be a linter error but looks like that's broken or misconfigured right now.